ocaml/xapi/message_forwarding.ml

```diff
@@ -3749,7 +3749,12 @@ module Forward = functor(Local: Custom_actions.CUSTOM_ACTIONS) -> struct
 				Helpers.call_api_functions ~__context
 					(fun rpc session_id ->
 						Client.SR.update rpc session_id sr
-					)
+					);
+
+				(* Attach SR-stats VDI if SR has stats capability *)
+				let sm_config = Db.SR.get_sm_config ~__context ~self:sr in
+				if (List.mem_assoc "SR_STATS" sm_config) then (* Need to confirm this condition and delete this comment *)
```

Comment:
The capabilities aren't stored in the `sm_config` field - you'll have to add support for the `SR_STATS` capability to [this file](https://github.com/xapi-project/xen-api/blob/master/ocaml/xapi/smint.ml) and then check whether the SR has the capability [like this](https://github.com/xapi-project/xen-api/blob/ec69fe9616c8a13c16b0b0e69f27f8854f7a402b/ocaml/xapi/xapi_vm_snapshot.ml#L189).